frontend/src/static/js/components/webstatus-overview-table.ts

```diff
@@ -147,122 +153,38 @@ export class WebstatusOverviewTable extends LitElement {
-          feature?.name.includes(searchValue))
-      ) {
-        features.push(feature);
-      }
-    }
-    return features;
-  }
-
-  reorderByQueryTerms(): components['schemas']['Feature'][] | undefined {
-    if (
-      !this.savedSearch ||
-      this.savedSearch.scope !== SavedSearchScope.GlobalSavedSearch ||
-      !this.savedSearch.value.is_ordered
-    ) {
-      return undefined;
-    }
-
-    const atoms: string[] = this.savedSearch.value.query.trim().split('OR');
-    const features = [];
-    for (const atom of atoms) {
-      const terms = atom.trim().split(':');
-      const foundFeatures = this.findFeaturesFromAtom(terms[0], terms[1]);
-      if (foundFeatures) {
-        features.push(...foundFeatures);
-      }
-    }
-
-    if (features.length !== this.taskTracker?.data?.data?.length) {
-      void new Toast().toast(
-        `Unable to apply custom sorting to saved search "${this.savedSearch.value.name}". Defaulting to normal sorting.`,
-        'warning',
-        'exclamation-triangle',
-      );
-      return undefined;
-    }
-    return features;
-  }
-
   render(): TemplateResult {
-    const columns: ColumnKey[] = parseColumnsSpec(
-      getColumnsSpec(this.location),
-    );
-    const sortSpec: string =
-      getSortSpec(this.location) || (DEFAULT_SORT_SPEC as string);
-
-    let headerCells: TemplateResult[] = [];
-    if (
-      this.savedSearch?.scope === SavedSearchScope.GlobalSavedSearch &&
-      this.savedSearch.value?.is_ordered
-    ) {
-      headerCells = renderSavedSearchHeaderCells(
-        this.savedSearch.value.name,
-        columns,
-      );
-    } else {
-      headerCells = columns.map(
-        col => html`${renderHeaderCell(this.location, col, sortSpec)}`,
-      );
-    }
-
     return html`
       <table class="data-table">
-        ${renderColgroups(columns)}
+        ${renderColgroups(this.columns)}
         <thead>
-          ${renderGroupsRow(columns)}
+          ${renderGroupsRow(this.columns)}
           <tr class="header-row">
-            ${headerCells}
+            ${this.headerCells}
           </tr>
         </thead>
         <tbody>
-          ${this.renderTableBody(columns)}
+          ${this.renderTableBody(this.columns)}
         </tbody>
       </table>
     `;
   }
 
   renderTableBody(columns: ColumnKey[]): TemplateResult {
-    switch (this.taskTracker.status) {
-      case TaskStatus.COMPLETE:
-        return this.taskTracker.data?.data?.length === 0
-          ? this.renderBodyWhenNoResults(columns)
-          : this.renderBodyWhenComplete(columns);
-      case TaskStatus.ERROR:
-        return this.renderBodyWhenError(columns);
-      case TaskStatus.INITIAL:
-        // Do the same thing as pending.
-        return this.renderBodyWhenPending(columns);
-      case TaskStatus.PENDING:
-        return this.renderBodyWhenPending(columns);
+    if (this.isLoading) {
+      return this.renderBodyWhenPending(columns);
+    }
+    if (this.data === undefined) {
```

Comment:
Any reason to not pass in the task tracker (instead of using `data`, dataError`and isLoading` as separate properties)?

I ask because it was not obvious at first that `this.data === undefined` means an error occurred. Whereas before, you could check the status.

Reply:
My thought process was to change this component to be more generic and only handle the data it needs to display, rather than the task to load the data. That might be misguided, so I'm open to other directions here. 🙂

Reply:
[TaskTracker](https://github.com/GoogleChrome/webstatus.dev/blob/dd93b719b2106c840f06e02bec5a91c9f1a6bb7a/frontend/src/static/js/utils/task-tracker.ts#L24) is meant to be something generic that lives outside Task. I always meant to go back and use a different enum from the  lit/task status enum. But as of right now, their current states work fine enough. But if we ever wanted to, we could easily use change the set of states to something that better reflects our user experience.

Reply:
I've made the change, and I agree it is the right choice. 🙂